tests/test_feature_aggregate.py

```diff
@@ -299,6 +312,26 @@ def test_aggregate_region_with_components(simple_df):
     assert _df.check_aggregate_region(v, components=["foo"]) is None
 
 
+@pytest.mark.parametrize(
+    "data, variable, weight",
+    ((NEG_WEIGHTS_DF, "Price|Carbon", "Emissions|CO2"),),
+)
+def test_agg_weight(data, variable, weight):
+
+    test_1 = (
+        IamDataFrame(data)
+        .aggregate_region(variable, weight=weight, drop_negative=False)
+        ._data
+    )
+    test_2 = (
+        IamDataFrame(data)
+        .aggregate_region(variable, weight=weight, drop_negative=True)
+        ._data
+    )
+
+    assert not test_1.equals(test_2)
```

Comment:
Here, it would be better to explicitly test some values, not just that they are not equal - they might be not equal for a lot of wrong reasons...